### ISSUE TITLE
cue/errors: a wrapped error list can't be printed as multiple errors

### ISSUE THREAD

**Issue by rogpeppe** [cue-lang/cue#1429]
Currently the `errors.Print` function will print multiple errors if the underlying error is directly produced by the `cue/errors` package, but not if such an error has been wrapped with `fmt.Errorf("%w", err)`.

Relatedly, various doc comments in the `errors` package refer to `List` as if it were an exported type when it actually isn't.

**Comment by myitcv** [comment]
Thanks @rogpeppe. For the sake of completeness, can you confirm you were working against tip 28b42576ce831209f4dd796d3515f16f784ba570?

Also do you have a small repro to hand?

**Comment by rogpeppe** [comment]
Yes, I was working against tip. This CL fixes the issue. The test in that CL failed before the code was changed: https://review.gerrithub.io/c/cue-lang/cue/+/529383

**Comment by myitcv** [comment]
Got it, thanks. Please can you update the comment in that CL to reference this issue?

https://github.com/cue-lang/cue/blob/master/doc/contribute.md#good-commit-messages

Something like:

```
...

Fixes #1429
```

should suffice to link it. We don't yet have a `gopherbot` like tool that adds a link from the issue to the CL. But when the CL trybots run it does sort of get linked.

**Comment by rogpeppe** [comment]
Done

**Comment by mvdan** [comment]
I don't know what happened here, but https://review.gerrithub.io/c/cue-lang/cue/+/529383 was merged without actually marking this issue as fixed. Doing so manually now, even if it's nearly four years later.
